packages/editor/src/store/selectors.js

```diff
@@ -1001,57 +1000,17 @@ export const getEditedPostContent = createRegistrySelector(
 	}
 );
 
-/**
- * Returns state object prior to a specified optimist transaction ID, or `null`
- * if the transaction corresponding to the given ID cannot be found.
- *
- * @param {Object} state         Current global application state.
- * @param {Object} transactionId Optimist transaction ID.
- *
- * @return {Object} Global application state prior to transaction.
- */
-export function getStateBeforeOptimisticTransaction( state, transactionId ) {
-	const transaction = find(
-		state.optimist,
-		( entry ) =>
-			entry.beforeState &&
-			get( entry.action, [ 'optimist', 'id' ] ) === transactionId
-	);
-
-	return transaction ? transaction.beforeState : null;
-}
-
 /**
  * Returns true if the post is being published, or false otherwise.
  *
  * @param {Object} state Global application state.
  *
  * @return {boolean} Whether post is being published.
  */
-export function isPublishingPost( state ) {
-	if ( ! isSavingPost( state ) ) {
-		return false;
-	}
-
-	// Saving is optimistic, so assume that current post would be marked as
-	// published if publishing
-	if ( ! isCurrentPostPublished( state ) ) {
-		return false;
-	}
-
-	// Use post update transaction ID to retrieve the state prior to the
-	// optimistic transaction
-	const stateBeforeRequest = getStateBeforeOptimisticTransaction(
-		state,
-		POST_UPDATE_TRANSACTION_ID
-	);
-
-	// Consider as publishing when current post prior to request was not
-	// considered published
-	return (
-		!! stateBeforeRequest &&
-		! isCurrentPostPublished( null, stateBeforeRequest.currentPost )
-	);
+export function isPublishingPost() {
+	// TODO: this selector always returns false
```

Comment:
This will be deprecated too, no?

Reply:
no, I kept it as is in this PR (always return false) but I want to see where it was used and what happens if it behave properly :P

Reply:
I checked that it was only used on Post publish button label, that's why I asked. I'll see your update then..

Reply:
I rebased and fixed the selector